Subject: [PATCH] LocalPiped: use local API URLs
---
Index: .env
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.env b/.env
--- a/.env	(revision 9733656066c1a14b7a9d3028680cbb3356114687)
+++ b/.env	(revision 21a18f71c7b3056b0581eef1d340ca579f724975)
@@ -1,3 +1,3 @@
-VITE_PIPED_API=https://pipedapi.kavin.rocks
-VITE_PIPED_PROXY=https://pipedproxy.kavin.rocks
+VITE_PIPED_API=http://localhost:8170
+VITE_PIPED_PROXY=http://localhost:8172
 VITE_PIPED_INSTANCES=https://piped-instances.kavin.rocks/
