Subject: [PATCH] LocalPiped: sqlite
---
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision 9f929cdf74f1c49d260e2d824cb72a60ab302ff7)
+++ b/build.gradle	(revision c29d0406d6c58e27456697daf61861ba9e0b6dbb)
@@ -30,8 +30,8 @@
     implementation 'io.activej:activej-specializer:5.5'
     implementation 'io.activej:activej-launchers-http:5.5'
     implementation 'org.hsqldb:hsqldb:2.7.3'
-    //implementation 'org.postgresql:postgresql:42.7.3'
-    implementation 'com.h2database:h2:2.3.232'
+    implementation 'org.xerial:sqlite-jdbc:3.46.1.3'
+    implementation 'org.hibernate.orm:hibernate-community-dialects:6.6.1.Final'
     implementation 'org.hibernate:hibernate-core:6.4.1.Final'
     implementation 'org.hibernate:hibernate-hikaricp:6.4.1.Final'
     implementation 'org.liquibase:liquibase-core:4.28.0'
Index: src/main/java/me/kavin/piped/consts/Constants.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/me/kavin/piped/consts/Constants.java b/src/main/java/me/kavin/piped/consts/Constants.java
--- a/src/main/java/me/kavin/piped/consts/Constants.java	(revision 9f929cdf74f1c49d260e2d824cb72a60ab302ff7)
+++ b/src/main/java/me/kavin/piped/consts/Constants.java	(revision c29d0406d6c58e27456697daf61861ba9e0b6dbb)
@@ -163,13 +163,10 @@
             } else {
                 S3_CLIENT = null;
             }
-            System.getenv().forEach((key, value) -> {
-                if (key.startsWith("hibernate"))
-                    hibernateProperties.put(key, value);
-            });
             MATRIX_SERVER = getProperty(prop, "MATRIX_SERVER", "https://matrix-client.matrix.org");
             MATRIX_TOKEN = getProperty(prop, "MATRIX_TOKEN");
             GEO_RESTRICTION_CHECKER_URL = getProperty(prop, "GEO_RESTRICTION_CHECKER_URL");
+
             prop.forEach((_key, _value) -> {
                 String key = String.valueOf(_key), value = String.valueOf(_value);
                 if (key.startsWith("hibernate"))
@@ -177,6 +174,11 @@
                 else if (key.startsWith("frontend."))
                     frontendProperties.put(StringUtils.substringAfter(key, "frontend."), value);
             });
+            System.getenv().forEach((key, value) -> {
+                if (key.startsWith("hibernate"))
+                    hibernateProperties.put(key.replaceAll("_", "."), value);
+            });
+
             frontendProperties.put("imageProxyUrl", IMAGE_PROXY_PART);
             frontendProperties.putArray("countries").addAll(
                     YOUTUBE_SERVICE.getSupportedCountries().stream().map(ContentCountry::getCountryCode)
Index: src/main/java/me/kavin/piped/utils/LiquibaseHelper.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/me/kavin/piped/utils/LiquibaseHelper.java b/src/main/java/me/kavin/piped/utils/LiquibaseHelper.java
--- a/src/main/java/me/kavin/piped/utils/LiquibaseHelper.java	(revision 9f929cdf74f1c49d260e2d824cb72a60ab302ff7)
+++ b/src/main/java/me/kavin/piped/utils/LiquibaseHelper.java	(revision c29d0406d6c58e27456697daf61861ba9e0b6dbb)
@@ -24,7 +24,7 @@
         String password = Constants.hibernateProperties.get("hibernate.connection.password");

         // ensure postgres driver is loaded
-        DriverManager.registerDriver(new org.h2.Driver());
+        DriverManager.registerDriver(new org.sqlite.JDBC());

         // register YugabyteDB database
         DatabaseFactory.getInstance().register(new liquibase.ext.yugabytedb.database.YugabyteDBDatabase());
Index: src/main/java/me/kavin/piped/utils/obj/db/Playlist.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/me/kavin/piped/utils/obj/db/Playlist.java b/src/main/java/me/kavin/piped/utils/obj/db/Playlist.java
--- a/src/main/java/me/kavin/piped/utils/obj/db/Playlist.java	(revision 9f929cdf74f1c49d260e2d824cb72a60ab302ff7)
+++ b/src/main/java/me/kavin/piped/utils/obj/db/Playlist.java	(revision c29d0406d6c58e27456697daf61861ba9e0b6dbb)
@@ -27,7 +27,7 @@
     private long id;

     @Column(name = "playlist_id", unique = true, nullable = false)
-    @GeneratedValue(generator = "UUID", strategy = GenerationType.IDENTITY)
+    //@GeneratedValue(generator = "UUID", strategy = GenerationType.IDENTITY)
     private UUID playlist_id;

     @Column(name = "name", length = 200)
Index: src/main/resources/hibernate.cfg.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/hibernate.cfg.xml b/src/main/resources/hibernate.cfg.xml
--- a/src/main/resources/hibernate.cfg.xml	(revision 9f929cdf74f1c49d260e2d824cb72a60ab302ff7)
+++ b/src/main/resources/hibernate.cfg.xml	(revision c29d0406d6c58e27456697daf61861ba9e0b6dbb)
@@ -4,7 +4,7 @@
         "http://www.hibernate.org/dtd/hibernate-configuration-3.0.dtd">
 <hibernate-configuration>
     <session-factory>
-        <property name="hibernate.hbm2ddl.auto">create</property>
+        <property name="hibernate.hbm2ddl.auto">update</property>
         <!-- Optional: Show SQL output for debugging -->
         <property name="hibernate.show_sql">false</property>
         <property name="hibernate.format_sql">true</property>
